### PR TITLE
Update spec for divide

### PR DESCRIPTION
**Pull request by pelszkow**
### Details:
Update specification for divide reference implementation

Depends on #4366 which introduce broadcast_rules.md.

### Tickets:
 - 46078

### CODE REVIEW

File: docs/ops/broadcast_rules.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Broadcast rules for elementwise ops {#openvino_docs_ops_broadcast_rules}
```
Comment (sdurawa) [inline docs/ops/broadcast_rules.md:1]: broadcast_rules.md is introduced with [4366](https://github.com/openvinotoolkit/openvino/pull/4366/), it is not required to review it in 2 different commits.

Comment (pelszkow) [reply docs/ops/broadcast_rules.md:1]: It's here to have good doc generations. I'll live it empty in this PR.

File: docs/ops/arithmetic/Divide_1.md [hunk 2]
```diff
@@ -6,46 +6,46 @@
...
   * **Type**: string
   * **Default value**: "numpy"
   * **Required**: *no*
 
 **Inputs**
 
-* **1**: A tensor of type T. **Required.**
-* **2**: A tensor of type T. **Required.**
+* **1**: A tensor of type T and arbitrary shape and rank. **Required.**
+* **2**: A tensor of type T and arbitrary shape and rank. Tensor element equalt to zerof for integer values lead to undefined behaviour. **Required.**
```
Comment (sdurawa) [inline docs/ops/arithmetic/Divide_1.md:31]: typeo equalt

File: docs/ops/arithmetic/Divide_1.md [hunk 3]
```diff
@@ -6,46 +6,46 @@
...
+
 **Attributes**:
 
 * *auto_broadcast*
 
   * **Description**: specifies rules used for auto-broadcasting of input tensors.
-  * **Range of values**:
-    * *none* - no auto-broadcasting is allowed, all input shapes should match
-    * *numpy* - numpy broadcasting rules, aligned with ONNX Broadcasting. Description is available in <a href="https://github.com/onnx/onnx/blob/master/docs/Broadcasting.md">ONNX docs</a>.
+  * **Range of values**: refer to rule #1 in broadcast rules,description is available in [Broadcast Rules For Elementwise Operations](../broadcast_rules.md).
```
Comment (sdurawa) [inline docs/ops/arithmetic/Divide_1.md:23]: please keep attributes explanation, for numpy refer to 'Broadcast Rules For Elementwise Operations'.

File: docs/ops/arithmetic/Divide_1.md (hunk 2 above)
Comment (pelszkow) [reply docs/ops/arithmetic/Divide_1.md:31]: fix

File: docs/ops/arithmetic/Divide_1.md (hunk 3 above)
Comment (pelszkow) [reply docs/ops/arithmetic/Divide_1.md]: fix

File: docs/ops/arithmetic/Divide_1.md [hunk 4]
```diff
@@ -6,46 +6,48 @@
...
 
 * *auto_broadcast*
 
   * **Description**: specifies rules used for auto-broadcasting of input tensors.
-  * **Range of values**:
-    * *none* - no auto-broadcasting is allowed, all input shapes should match
-    * *numpy* - numpy broadcasting rules, aligned with ONNX Broadcasting. Description is available in <a href="https://github.com/onnx/onnx/blob/master/docs/Broadcasting.md">ONNX docs</a>.
+  * **Range of values**:  
+    * *none* - no auto-broadcasting is allowed, all input shapes must match,
+    * *numpy* - numpy broadcasting rules, description is available in [Broadcast Rules For Elementwise Operations](../broadcast_rules.md).
```
Comment (sdurawa) [inline docs/ops/arithmetic/Divide_1.md:25]: add `pdpd` attribute.

File: docs/ops/arithmetic/Divide_1.md [hunk 5]
```diff
@@ -4,48 +4,51 @@
...
+**Short description**: *Divide* performs element-wise division operation with two given tensors applying broadcasting rule specified in the *auto_broacast* attribute.
+
+**Detailed description**
+Before performing arithmetic operation, input tensors *a* and *b* are broadcasted if their shapes are different and `auto_broadcast` attribute is not `none`. Broadcasting is performed according to `auto_broadcast` value.
+
+After broadcasting *Divide* performs divison operation for the input tensors *a* and *b* using the formula below:
+
+\f[
+o_{i} = a_{i} / b_{i}
+\f]
```
Comment (lazarevevgeny) [inline docs/ops/arithmetic/Divide_1.md:16]: add a comment about division by zero. Refer to FloorMod and Mod for the example.

**Review by lazarevevgeny — CHANGES_REQUESTED** [review]

File: docs/ops/arithmetic/Divide_1.md (hunk 5 above)
Comment (pelszkow) [reply docs/ops/arithmetic/Divide_1.md:16]: done

File: docs/ops/arithmetic/Divide_1.md [hunk 6]
```diff
@@ -4,48 +4,63 @@
...
-**Detailed description**
-Before performing arithmetic operation, input tensors *a* and *b* are broadcasted if their shapes are different and `auto_broadcast` attributes is not `none`. Broadcasting is performed according to `auto_broadcast` value.
-
-After broadcasting *Divide* does the following with the input tensors *a* and *b*:
-
-\f[
-o_{i} = a_{i} / b_{i}
-\f]
 
 **Examples**
```
Comment (sdurawa) [inline docs/ops/arithmetic/Divide_1.md:57]: missing example with `pythondiv` attribute

Comment (pelszkow) [reply docs/ops/arithmetic/Divide_1.md:57]: Do you expect something like this:
```
*Example 1*

<layer ... type="Divide">
    <data m_pythondiv=true />
    <input>
        <port id="0">
            <dim>256</dim>
            <dim>56</dim>
        </port>
        <port id="1">
            <dim>256</dim>
            <dim>56</dim>
        </port>
    </input>
    <output>
        <port id="2">
            <dim>256</dim>
            <dim>56</dim>
        </port>
    </output>
</layer>
```

File: docs/ops/arithmetic/Divide_1.md [hunk 7]
```diff
@@ -4,48 +4,63 @@
...
+**Short description**: *Divide* performs element-wise division operation with two given tensors applying broadcasting rule specified in the *auto_broacast* attribute.
+
+**Detailed description**
+Before performing arithmetic operation, input tensors *a* and *b* are broadcasted if their shapes are different and `auto_broadcast` attribute is not `none`. Broadcasting is performed according to `auto_broadcast` value.
+After broadcasting *Divide* performs division operation for the input tensors *a* and *b* using the formula below:
+
+\f[
+o_{i} = a_{i} / b_{i}
+\f]
+
```
Comment (lazarevevgeny) [inline docs/ops/arithmetic/Divide_1.md:16]: Remove redundant empty line.

File: docs/ops/arithmetic/Divide_1.md (hunk 6 above)
Comment (sdurawa) [reply docs/ops/arithmetic/Divide_1.md:57]: I think we can add this attribute into existing examples, so we have both attributes:
`<data auto_broadcast="none" m_pythondiv="true"/>`

Comment (pelszkow) [reply docs/ops/arithmetic/Divide_1.md:57]: done

File: docs/ops/arithmetic/Divide_1.md (hunk 7 above)
Comment (pelszkow) [reply docs/ops/arithmetic/Divide_1.md:16]: done

**Comment by pelszkow** [comment]
All suggestion applied. Do you see something more to fix or it can be merge?

**Review by lazarevevgeny — APPROVED** [review]

**Review by sdurawa — APPROVED** [review]

**Review by ilyachur — APPROVED** [review]

**Pull request merged**